pmd-java/src/main/java/net/sourceforge/pmd/lang/java/ast/ASTModuleDirective.java

```diff
@@ -4,44 +4,47 @@
 
 package net.sourceforge.pmd.lang.java.ast;
 
-public final class ASTModuleDirective extends AbstractJavaNode {
 
-    public enum DirectiveType {
-        REQUIRES, EXPORTS, OPENS, USES, PROVIDES
-    }
+import net.sourceforge.pmd.lang.rule.xpath.NoAttribute;
 
-    public enum RequiresModifier {
-        STATIC, TRANSITIVE
-    }
+/**
+ * A directive of a {@linkplain ASTModuleDeclaration module declaration}.
+ * Implementations provide more specific attributes.
+ *
+ * <pre class="grammar">
+ *
+ * ModuleDirective ::= {@linkplain ASTModuleRequiresDirective ModuleRequiresDirective}
+ *                   | {@linkplain ASTModuleOpensDirective ModuleOpensDirective}
+ *                   | {@linkplain ASTModuleExportsDirective ModuleExportsDirective}
+ *                   | {@linkplain ASTModuleProvidesDirective ModuleProvidesDirective}
+ *                   | {@linkplain ASTModuleUsesDirective ModuleUsesDirective}
+ *
+ * </pre>
+ */
+public abstract class ASTModuleDirective extends AbstractJavaNode {
 
-    private DirectiveType type;
+    private final DirectiveType type;
 
-    private RequiresModifier requiresModifier;
 
-    ASTModuleDirective(int id) {
+    ASTModuleDirective(int id, DirectiveType type) {
         super(id);
-    }
-
-
-    @Override
-    protected <P, R> R acceptVisitor(JavaVisitor<? super P, ? extends R> visitor, P data) {
-        return visitor.visit(this, data);
-    }
-
-
-    void setType(DirectiveType type) {
         this.type = type;
     }
 
-    public String getType() {
-        return String.valueOf(type);
+    /**
+     * Returns the kind of the directive.
+     */
+    @NoAttribute
+    public DirectiveType getType() {
+        return type;
     }
 
-    void setRequiresModifier(RequiresModifier requiresModifier) {
-        this.requiresModifier = requiresModifier;
+    /**
+     * Kind of a module directive. Specific kinds are represented by
+     * specific subclasses.
```

Comment:
Do we still need this enum then when we have the subclasses?

Reply:
Removing it